core/Date.php

```diff
@@ -611,6 +612,38 @@ public function subYear($n)
         return new Date($result, $this->timezone);
     }
 
+    protected static $use12HourClock;
+
+    protected function getTimeFormat()
+    {
+        if (is_null(self::$use12HourClock)) {
+            self::$use12HourClock = LanguagesManager::uses12HourClockForCurrentUser();
```

Comment:
Tried to implement something similar now (see db969cd), feels better that way :)